day1/es3/es3.qml

```diff
@@ -0,0 +1,82 @@
+import QtQuick 2.12
+import QtQuick.Window 2.12
+
+Window {
+    width: 640
+    height: 800
+    visible: true
+    title: qsTr("Chapters exercise")
+
+    Column {
+        id: allChapters
+
+        readonly property int animationTime: 300
+
+        function changeChapter(clickedChapter){
```

Comment:
In generale in QML più si evita di usare codice Javascript in questo modo e meglio è.
Questo tipo di codice è estremamente difficile da manutenere, specialmente in un progetto grande nel lungo periodo quando tante persone ci mettono le mani, ed è anche difficile da validare formalmente come corretto a prima vista essendo Javascript un linguaggio non fortemente tipizzato.

L'ideale è sempre preferire codice dichiarativo quando si usa il QML, e lasciare il codice imperativo più al lato C++.

Il mio consiglio è di provare così:
- definisci un `ListModel` con all'interno 3 `ListElement`, ciascuno contentente il `content` testuale che dovrà finire nei `Chapter`
- usa un componente `ListView` per generare una lista, e assegnagli il `ListModel`
- usa i componenti `Chapter` come delegate della `ListView`

A questo punto lascerei a te escogitare qualcosa per trasformare questa funzione `changeChapter(...)` in codice dichiarativo :)

Vedrai che alla fine viene molto meno codice, più semplice da leggere e di conseguenza da manutere

Reply:
Ho fatto una versione con codice javascirpt ridotto, ma usa  _"Column"_ , si trova nel commit _"reduced to minimum use of javascript"_, dopo ho fatto la versione con _"ListView"_, in questo caso ho dovuto aggiungere un po' più di codice rispetto alla versione senza _"ListView"_, si trova nel commit _"simplified a little othersChaptersPermitOpening"_